src/compute/src/arrangement/manager.rs

```diff
@@ -225,4 +155,131 @@ impl TraceBundle {
             &self.errs.get_logical_compaction(),
         )
     }
+
+    /// Turns this trace bundle into a padded version that reports empty data for all times less
+    /// than the traces' current logical compaction frontier.
+    ///
+    /// Note that the padded bundle represents a different TVC than the original one, it is unsound
+    /// to use it to "uncompact" an existing TVC. The only valid use of the padded bundle is to
+    /// initializa a new TVC.
+    pub fn into_padded(self) -> Self {
+        Self {
+            oks: self.oks.into_padded(),
+            errs: self.errs.into_padded(),
+            to_drop: self.to_drop,
+        }
+    }
+}
+
+/// Handle to a trace managed by the `TraceManager`.
+///
+/// A trace can be optionally padded. A padded trace contains empty data for all times greater than
+/// or equal to its `padded_since` and less than the logical compaction frontier of the inner
+/// `trace`.
+#[derive(Clone)]
+pub struct TraceHandle<Tr> {
```

Comment:
How does this compare to differential's `TraceFrontier` type?

Reply:
From what I can tell `TraceFrontier` caps the inner trace's frontiers by the provided `since` and `upper`, but doesn't perform any padding as this type does. A `TraceFrontier::since` that's before the inner trace's compaction frontier has no effect.